src/controllers/authentication.ts

```diff
@@ -0,0 +1,53 @@
+import jwt, { JwtPayload } from 'jsonwebtoken'
+import express from 'express'
+import User from '../database/models/userModel'
+import Token from '../database/models/tokenModel'
+
+export const validateAndRetrieveToken = async (token: string) => {
+  try {
+    const decodedToken = jwt.verify(token, process.env.JWT_SECRET, {
+      ignoreExpiration: false,
+    }) as JwtPayload
+    const myToken = await Token.findOne({ where: { token } })
+
+    if (!myToken) {
+      return null
+    }
+
+    return decodedToken
+  } catch (error) {
+    return null
+  }
+}
+
+export const verifyEmail = async (
+  req: express.Request,
+  res: express.Response,
+) => {
+  try {
+    const { token } = req.params
```

Comment:
let us have this function into user controller, and try to call validateAndRetrieveToken(token) to verify the token.